File: api_nestjs_distributed_order_platform/src/application/port/client.repository.ts
import { ClientEntity } from "src/domain/entities/client.entity";

export interface ClientRepositoryInterface {
    create(client: ClientEntity): Promise<ClientEntity>;
    findById(clientId: string): Promise<ClientEntity | null>;
    findAll(): Promise<ClientEntity[]>;
    update(client: ClientEntity): Promise<void>;
    delete(clientId: string): Promise<void>;
}

File: api_java_distributed_order_platform/src/main/java/com/project/order/application/port/ClientRepository.java
package com.project.order.application.port;

import com.project.order.domain.model.Client;

import java.util.List;
import java.util.Optional;

public interface ClientRepository {
    Client create(Client client);
    Optional<Client> findById(String id);
    List<Client> findAll();
    Client update(Client client);
    void delete(String id);
}
